rust/src/dataset/scanner.rs

```diff
@@ -19,17 +19,29 @@ use std::sync::Arc;
 
 use arrow_array::RecordBatch;
 use arrow_schema::{Schema as ArrowSchema, SchemaRef};
+use futures::stream::Stream;
+use object_store::path::Path;
+use tokio::sync::mpsc::{self, Receiver};
 
 use super::Dataset;
 use crate::datatypes::Schema;
-use crate::error::Result;
-use crate::format::Fragment;
-use crate::io::FileReader;
+use crate::format::{Fragment, Manifest};
+use crate::io::{FileReader, ObjectStore};
+use crate::Result;
 
 /// Dataset Scanner
 ///
-/// TODO: RecordBatchReader does not support async iterator yet
-/// we need to use a tokio::rt::block_on for the iterator.
+/// ```rust,ignore
+/// let dataset = Dataset::open(uri).await.unwrap();
+/// let stream = dataset.scan()
+///     .project(&["col", "col2.subfield"]).unwrap()
+///     .limit(10)
```

Comment:
Would it make sense to change this and projection to accept Option? Usually from the incoming code these are optional parameters

Reply:
`project(Option<Vec<>>)` ?

Should we only call `project()` if the parameter is `Some(...)`?